### PR TITLE
metrics:fix add index duration bucket dosen't work

### PR DESCRIPTION
**Pull request by ciscoxll**
See in grafana batch add index dosen't work.

`histogram_quantile(1, sum(rate(tidb_ddl_batch_add_idx_succ_bucket[1m])) by (le, type))`

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -751,6 +751,14 @@ func (d *ddl) backfillBatchTasks(startTime time.Time, startHandle int64, reorgIn
 	// nextHandle will be updated periodically in runReorgJob, so no need to update it here.
 	d.reorgCtx.setNextHandle(nextHandle)
 	metrics.BatchAddIdxHistogram.Observe(elapsedTime)
+  
+  metrics.JobsGauge.WithLabelValues(reorgInfo.Type.String()).Inc()
+	timeStart := time.Now()
+	defer func() {
+		metrics.JobsGauge.WithLabelValues(reorgInfo.Type.String()).Dec()
+		metrics.BatchAddIdxHistogram.WithLabelValues(reorgInfo.Type.String(), metrics.RetLabel(err)).Observe(time.Since(timeStart).Seconds())
```
Comment (shenli) [inline ddl/index.go]: So the observed time is always zero, right?

**Comment by shenli** [comment]
Why not modify the grafana config to show the metrics of BatchAddIdxHistogram?

File: ddl/index.go (hunk 1 above)
Comment (ciscoxll) [reply ddl/index.go]: Observed always no data points.

**Comment by winkyao** [comment]
@ciscoxll You can modify the grafana config in https://github.com/pingcap/tidb-ansible/blob/311eef2058f8fb18e7971e0247dbd9ef7dd12d77/scripts/tidb.json

**Comment by ciscoxll** [comment]
@winkyao If you change `tidb.json`, both `tidb`and`tidb.json` need to be modified, so I just changed tidb.

**Comment by ciscoxll** [comment]
@zimulala @winkyao PTAL

File: ddl/index.go [hunk 2]
```diff
@@ -746,14 +746,16 @@ func (d *ddl) backfillBatchTasks(startTime time.Time, startHandle int64, reorgIn
 		err1 := kv.RunInNewTxn(d.store, true, func(txn kv.Transaction) error {
 			return errors.Trace(reorgInfo.UpdateHandle(txn, nextHandle))
 		})
+		metrics.BatchAddIdxHistogram.WithLabelValues(metrics.BatchAddIndexFail, metrics.RetLabel(err)).Observe(elapsedTime)
```
Comment (shenli) [inline ddl/index.go]: err is not nil, why use metrics.RetLabel(err)?

Comment (ciscoxll) [reply ddl/index.go]: @shenli RetLabel returns "ok" when err == nil and "err" when err != nil, so I can use it.

File: metrics/ddl.go [hunk 3]
```diff
@@ -34,14 +34,16 @@ var (
...
+	BatchAddIdxHistogram = prometheus.NewHistogramVec(
 		prometheus.HistogramOpts{
 			Namespace: "tidb",
 			Subsystem: "ddl",
 			Name:      "batch_add_idx_duration_seconds",
 			Help:      "Bucketed histogram of processing time (s) of batch handle data",
-			Buckets:   prometheus.ExponentialBuckets(0.001, 2, 20),
-		})
+			Buckets:   prometheus.ExponentialBuckets(0.01, 2, 20),
+		}, []string{LblType, LblResult})
```
Comment (zimulala) [inline metrics/ddl.go]: We already have `LblResult`. Why we need `LblType`?

File: metrics/ddl.go [hunk 4]
```diff
@@ -34,14 +34,16 @@ var (
...
+	BatchAddIndexFail    = "batch_add_idx_fail"
+	BatchAddIdxHistogram = prometheus.NewHistogramVec(
 		prometheus.HistogramOpts{
 			Namespace: "tidb",
 			Subsystem: "ddl",
 			Name:      "batch_add_idx_duration_seconds",
 			Help:      "Bucketed histogram of processing time (s) of batch handle data",
-			Buckets:   prometheus.ExponentialBuckets(0.001, 2, 20),
-		})
+			Buckets:   prometheus.ExponentialBuckets(0.01, 2, 20),
```
Comment (zimulala) [inline metrics/ddl.go]: Why change it?

File: ddl/index.go (hunk 2 above)
Comment (zimulala) [reply ddl/index.go]: If we can remove `LblResult`.
I think we can put this code on line 743. Then we can remove line 757.

**Comment by ciscoxll** [comment]
@zimulala  @coocood  PTAL

File: metrics/ddl.go [hunk 5]
```diff
@@ -34,14 +34,16 @@ var (
 			Buckets:   prometheus.ExponentialBuckets(0.01, 2, 20),
 		}, []string{LblType, LblResult})
 
-	BatchAddIdxHistogram = prometheus.NewHistogram(
+	BatchAddIndexSucc    = "batch_add_idx_succ"
```
Comment (coocood) [inline metrics/ddl.go]: Can we just use `LblOK` and `LblError`?

Comment (ciscoxll) [reply metrics/ddl.go]: @coocood If we use `LblOK` and `LblError`, It causes redeclared.

**Comment by ciscoxll** [comment]
@coocood PTAL

**Comment by shenli** [comment]
LGTM

**Pull request merged**